### PR TITLE
chore: fix setup dev script & DB alert

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

- Fix the `setup-dev` script to not remove `yarn.lock` to avoid the following error introduced by downstream package
```
$ tsc
../../node_modules/@types/babel__traverse/index.d.ts(297,72): error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'NodePath<T>' is not assignable to type 'NodePath<Node>'.
    Types of property 'hasNode' are incompatible.
      Type '() => this is NodePath<NonNullable<T>>' is not assignable to type '() => this is NodePath<Node>'.
        Type predicate 'this is NodePath<NonNullable<T>>' is not assignable to 'this is NodePath<Node>'. 
```
- Fix dependent bot alert https://github.com/aws-amplify/amplify-codegen/security/dependabot/33

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -107,7 +107,8 @@
     "cross-fetch": "^2.2.6",
     "glob-parent": "^6.0.2",
     "parse-url": "^8.1.0",
-    "graphql": "15.8.0"
+    "graphql": "15.8.0",
+    "xml2js": "0.5.0"
```
Comment (alharris-at) [inline package.json:111]: Why is the new dep necessary?

File: package.json [hunk 2]
```diff
@@ -23,8 +23,8 @@
     "verdaccio-publish": "yarn verdaccio-clean && yarn publish-to-verdaccio",
     "verdaccio-disconnect": "source .circleci/local_publish_helpers.sh && unsetNpmRegistryUrl",
     "verdaccio-stop": "kill -9 $(lsof -n -t -iTCP:4873 -sTCP:LISTEN)",
-    "setup-dev": "rm -rf yarn.lock && (yarn && lerna run build) && yarn add-cli-no-save && (yarn hoist-cli && yarn rm-dev-link && yarn link-dev)",
-    "add-cli-no-save": "yarn add @aws-amplify/amplify-category-api @aws-amplify/cli-internal -W && git restore package.json yarn.lock",
```
Comment (alharris-at) [inline package.json:27]: Did you verify that w/ these changes `setup-dev` still behaves as expected?

Comment (AaronZyLee) [reply package.json:27]: Yes I just verified locally. The new script works well.

File: package.json (hunk 1 above)
Comment (AaronZyLee) [reply package.json:111]: It is one of the downstream dependency of `aws-sdk`. The alternative is to bump `aws-sdk` instead

Comment (alharris-at) [reply package.json:111]: Oh I didn't realize this was in `resolutions`. Got it.

**Review by alharris-at — APPROVED** [review]

**Pull request merged**